pandas/core/indexers/objects.py

```diff
@@ -41,17 +41,33 @@
 
 
 class BaseIndexer:
-    """Base class for window bounds calculations."""
+    """
+    Base class for window bounds calculations.
+
+    Examples
+    --------
+    >>> from pandas.api.indexers import BaseIndexer
+    >>> class CustomIndexer(BaseIndexer):
+    ...     def get_window_bounds(self, num_values, min_periods, center, closed, step):
+    ...         start = np.empty(num_values, dtype=np.int64)
+    ...         end = np.empty(num_values, dtype=np.int64)
+    ...         for i in range(num_values):
+    ...             end[i] = i + self.window_size
+    ...         return start, end
+    >>> df = pd.DataFrame({"values": range(5)})
+    >>> indexer = CustomIndexer(window_size=2)
+    >>> df.rolling(indexer).sum()  # doctest: +SKIP
```

Comment:
what was it complaining about? could you show the error please?

Reply:
```(pandas-dev) *[Jul25][~/pandas-dealeon]$ ./scripts/validate_docstrings.py pandas.api.indexers.BaseIndexer
+ /Users/dealeon/mambaforge/envs/pandas-dev/bin/ninja
[1/1] Generating write_version_file with a custom command
[1]    33979 segmentation fault  ./scripts/validate_docstrings.py pandas.api.indexers.BaseIndexer```

Reply:
you need to update `start` inside the loop too (not just `end`)

Reply:
You are telling me that I don't know how to copy (examples that are already made)? You are correct.